examples/simulation/plot_simulate_sample_dataset.py

```diff
@@ -0,0 +1,161 @@
+    n_vertices = sum(len(v) for v in vertices)
+    del offsets
+    dotter = dotter[used]
+    assert dotter.shape[0] == n_vertices
+    print('    Reduced from %d sources to %d for %d labels'
+          % (len(used), used.sum(), len(label_names)))
+
+    # Actually do the generation
+    count = 0
+    tstep = 1. / raw.info['sfreq']
+    assert events[0, 0] > raw.first_samp
+    rev_id = {val: key for key, val in event_id.items()}
+    # one signal for all for now
+    n_active = int(round(0.2 * raw.info['sfreq']))
+    signal = 1e-9 * np.hanning(n_active)
+    while True:
+        if count == 0:
+            n_samp = events[0, 0] - raw.first_samp
+            id_ = 0
+        else:
+            if count == len(events):
+                n_samp = len(raw.times) - (events[-1, 0] - raw.first_samp)
+            else:
+                assert 0 < count < len(events)
+                n_samp = events[count, 0] - events[count - 1, 0]
+            id_ = events[count - 1, 2]
+        assert n_samp > 0
+        data = np.zeros((len(label_names), n_samp))
+        if id_ not in rev_id or rev_id[id_] not in activations:
+            print('    Ignoring event %d / %d (unknown id: %d)'
+                  % (count, len(events), id_))
+        else:
+            key = rev_id[id_]
+            print('    Generating STC for event %d / %d (type %s)'
+                  % (count, len(events), key))
+            for name, amp in activations[key]:
+                row_idx = label_names.index(name)
+                time_sl = slice(0, min(len(signal), data.shape[1]))
+                data[row_idx, time_sl] = amp * signal[time_sl]
+        # Project label activations into source space
+        data = np.dot(dotter, data)
+        # Yield the SourceEstimate
+        yield (mne.SourceEstimate(data, vertices, 0, tstep), id_)
+        count += 1
+
+
+##############################################################################
+# Run the simulation
+# ------------------
+
+# Compute a noise covariance to add noise to the data
+noise_epochs = mne.Epochs(raw, events, tmax=0)
+noise_epochs.info['bads'] = []
+noise_cov = mne.compute_covariance(noise_epochs)
+raw_sim = mne.simulation.simulate_raw(
+    raw, stc_gen(activations), None, None, None, noise_cov,
+    blink=True, ecg=True, forward=forward, random_state=0, verbose=True)
+
+##############################################################################
+# Source localize
+# ---------------
+
+method, lambda2 = 'dSPM', 1. / 9.
+epochs = mne.Epochs(raw, events, event_id)
+inv = mne.minimum_norm.make_inverse_operator(epochs.info, forward, noise_cov)
+stc_aud = mne.minimum_norm.apply_inverse(
+    epochs['auditory/left'].average(), inv, lambda2, method)
+stc_vis = mne.minimum_norm.apply_inverse(
+    epochs['visual/right'].average(), inv, lambda2, method)
+stc_diff = stc_aud - stc_vis
+brain = stc_diff.plot(subjects_dir=subjects_dir, initial_time=0.1)
```

Comment:
Output:
```
...
    Ignoring event 80 / 86 (unknown id: 5)
    Ignoring event 81 / 86 (unknown id: 32)
    Generating STC for event 82 / 86 (type visual/left)
    Generating STC for event 83 / 86 (type auditory/left)
    Generating STC for event 84 / 86 (type visual/right)
    Generating STC for event 85 / 86 (type auditory/right)
    Generating STC for event 86 / 86 (type visual/left)
Done
```
And image, as expected:

![Screenshot from 2019-03-12 15-24-41](https://user-images.githubusercontent.com/2365790/54229601-faa22380-44da-11e9-9fbd-da634e68a5a8.png)